### DISCUSSION TITLE
Support for more docker resources in terraform

### DISCUSSION THREAD

**Discussion by Lazzu** [renovatebot/renovate#26783] [Suggest an Idea]
Sometimes it is good practice to first fetch the image if it has been updated and then use it to define a container:
```tf
data "docker_registry_image" "uptimekuma" {
  name = "louislam/uptime-kuma:1.23.2-alpine"
}

resource "docker_image" "uptimekuma" {
  name          = data.docker_registry_image.uptimekuma.name
  pull_triggers = [data.docker_registry_image.uptimekuma.sha256_digest]
}

resource "docker_container" "uptimekuma" {
    name  = "uptimekuma"
    image = docker_image.uptimekuma.image_id
    /* ... */
}
```
Renovate does not seem to support `docker_registry_image` nor `docker_image` even though the docs say this:

> Renovate can update image references of the Docker provider resources (docker\_\*).

I wonder if it could be a possibility to add support for these?

**Comment by viceice** [comment]
sure, PR welcome

**Comment by davidw-verbund** [comment]
This might have changed since this post was made, but I can confirm that docker_image does work. I think Renovate is skipping it because it contains a variable similar to [this test](https://github.com/renovatebot/renovate/blob/8f29e5750e7e46fbeb3c05c542fb41d50853f679/lib/modules/manager/terraform/extract.spec.ts#L439C11-L439C42).

Anyway, I would also appreciate support for the docker_registry_image data source as it actually does not require a docker deamon since the latest update of the kreuzwerker/docker provider.

**Comment by interad-woergoetter** [comment]
I created a PR for adding support to the docker_registry_image data source: #35537

**Reply by interad-woergoetter** [reply]
Merged and available as release 40.25.0
